subsys/net/l2/ethernet/bridge.c

```diff
@@ -165,79 +220,247 @@ static inline bool is_link_local_addr(struct net_eth_addr *addr)
 	return false;
 }
 
-enum net_verdict net_eth_bridge_input(struct ethernet_context *ctx,
-				      struct net_pkt *pkt)
+static void random_linkaddr(uint8_t *linkaddr, size_t len)
+{
+	sys_rand_get(linkaddr, len);
+}
+
+static void bridge_iface_init(struct net_if *iface)
 {
-	struct eth_bridge *br = ctx->bridge.instance;
-	sys_snode_t *node;
+	struct eth_bridge_iface_context *ctx = net_if_get_device(iface)->data;
+	struct virtual_interface_context *vctx = net_if_l2_data(iface);
+	char name[MAX_BRIDGE_NAME_LEN];
 
-	NET_DBG("new pkt %p", pkt);
+	if (ctx->is_init) {
+		return;
+	}
+
+	k_mutex_init(&ctx->lock);
+
+	ctx->iface = iface;
+
+	net_if_flag_set(iface, NET_IF_NO_AUTO_START);
+	net_if_flag_clear(iface, NET_IF_IPV4);
+	net_if_flag_clear(iface, NET_IF_IPV6);
+	net_if_flag_clear(iface, NET_IF_FORWARD_MULTICASTS);
+
+	snprintk(name, sizeof(name), "bridge%d", ctx->id);
+	net_if_set_name(iface, name);
+
+	net_virtual_set_name(iface, "<no config>");
+
+	/* We need to set the link address here as it would be set in
+	 * attach function normally but we do not use that in bridging.
+	 */
+	random_linkaddr(vctx->lladdr.addr, sizeof(vctx->lladdr.addr));
+
+	vctx->lladdr.len = sizeof(vctx->lladdr.addr);
+	vctx->lladdr.type = NET_LINK_UNKNOWN;
+
+	net_if_set_link_addr(iface, vctx->lladdr.addr,
+			     vctx->lladdr.len, vctx->lladdr.type);
+
+	ctx->is_init = true;
+	ctx->is_setup = false;
+}
+
+static enum virtual_interface_caps bridge_get_capabilities(struct net_if *iface)
+{
+	ARG_UNUSED(iface);
+
+	return VIRTUAL_INTERFACE_BRIDGE;
+}
+
+static int bridge_iface_start(const struct device *dev)
+{
+	struct eth_bridge_iface_context *ctx = dev->data;
+
+	if (!ctx->is_setup) {
+		NET_DBG("Bridge interface %d not configured yet.",
+			net_if_get_by_iface(ctx->iface));
+		return -ENOENT;
+	}
+
+	if (ctx->status) {
+		return -EALREADY;
+	}
+
+	ctx->status = true;
+
+	NET_DBG("Starting iface %d", net_if_get_by_iface(ctx->iface));
+
+	/* You can implement here any special action that is needed
```

Comment:
This comment seems a bit odd, I see it's copied from the sample code where it makes more sense, shall it be removed?

Reply:
Sure, will remove